xscope_fileio/src/xscope_io_device.c

```diff
@@ -33,14 +34,23 @@ void xscope_fileio_lock_release(void) {
     lock_release(file_access_lock);
 }
 
+unsigned xscope_fileio_is_initialized(void) {
+    return xscope_io_init_flag;
+}
+
 void xscope_io_init(chanend_t xscope_end){
     xscope_fileio_lock_alloc();
     xscope_mode_lossless();
     c_xscope = xscope_end;
     xscope_connect_data_from_host(c_xscope);
+    xscope_io_init_flag = 1;
 }
 
 xscope_file_t xscope_open_file(const char* filename, char* attributes){
+    /* Wait until xscope_fileio is initialized */
+    while(xscope_fileio_is_initialized() == 0) {
+        delay_ticks(1);
```

Comment:
This won't actually implement an event based delay because the timer will always be ready to fire. It will work just fine though as a busy waiting delay.

Reply:
I intentionally did not use a hwtimer, as I didn't want to dictate hardware resource usage, if not necessary.

If an application wants to prevent busy waiting, it can use the new API xscope_fileio_is_initialized() and event based wait based on that result, thus allowing the application to decide event based or busy.